integration-healthcheck/src/main/java/org/symphonyoss/integration/healthcheck/IntegrationBridgeHealthAggregator.java

```diff
@@ -0,0 +1,138 @@
+   */
+  private static final String SUCCESS_MESSAGE = "Success";
+
+  /**
+   * Unknown version message
+   */
+  private static final String UNKNOWN_VERSION = "Unknown Version";
+
+  /**
+   * Version field
+   */
+  private static final String VERSION = "version";
+
+  /**
+   * Message field
+   */
+  private static final String MESSAGE = "message";
+
+  /**
+   * Integration Bridge version
+   */
+  private String bridgeVersion;
+
+  public IntegrationBridgeHealthAggregator() {
+    this.bridgeVersion = BRIDGE_VERSION == null ? UNKNOWN_VERSION : BRIDGE_VERSION;
+  }
+
+  @Override
+  public Health aggregate(Map<String, Health> healths) {
+    Health healthConnectivity = healths.get(CONNECTIVITY);
+    Health healthApplications = healths.get(APPLICATIONS);
+
+    Health.Builder builder = retrieveIntegrationBridgeStatus(healthApplications, healthConnectivity);
+
+    IntegrationBridgeHealthConnectivity connectivity =
+        (IntegrationBridgeHealthConnectivity) healthConnectivity.getDetails().get(CONNECTIVITY);
+    List<IntegrationHealth> appsHealth = getApplicationsHealth(healthApplications);
+
+    return builder.withDetail(VERSION, bridgeVersion).withDetail(CONNECTIVITY, connectivity)
+        .withDetail(APPLICATIONS, appsHealth).build();
+  }
+
+  /**
+   * Retrieves Integration Bridge main status with the rule:
+   * If at least one integration is "active", and connectivity with Agent, KM and POD is up, the
+   * main status for the Integration Bridge will be set to "UP". Otherwise, it will be set to
+   * "DOWN".
+   */
+  private Health.Builder retrieveIntegrationBridgeStatus(Health healthApplications, Health healthConnectivity) {
+    if (Status.DOWN.equals(healthApplications.getStatus())) {
+      // if all integrations are in a non-active status, Integration Bridge status should be "down".
+      return down("There is no active Integration");
+    }
+
+    if (Status.DOWN.equals(healthConnectivity.getStatus())) {
+      // if connectivity is down for Agent, KM or POD, Integration Bridge status should be "down".
+      return down("Connectivity is down for Agent, KM or POD");
+    }
+
+    return up();
+  }
+
+  /**
+   * Set the health status to UP
+   */
+  private Health.Builder up() {
+    return Health.up().withDetail(MESSAGE, SUCCESS_MESSAGE);
+  }
+
+  /**
+   * Set the health status to DOWN
+   */
+  private Health.Builder down(String message) {
+    return Health.down().withDetail(MESSAGE, message);
+  }
+
+  /**
+   * Retrieves the application statuses.
+   * @param healthApplications Health aggregator from all the deployed applications
+   * @return Application statuses.
+   */
+  private List<IntegrationHealth> getApplicationsHealth(Health healthApplications) {
+    List<IntegrationHealth> appsHealth = new ArrayList<>();
+    Map<String, Object> details = healthApplications.getDetails();
+
+    for (String key : details.keySet()) {
+      IntegrationHealth health = (IntegrationHealth) details.get(key);
```

Comment:
It's more efficient to use "entrySet" when iterating over a Map if you need "key" and "value".